core/src/main/kotlin/com/github/inflab/spring/data/mongodb/core/aggregation/search/QueryStringQueryOptionDsl.kt

```diff
@@ -41,7 +43,23 @@ class QueryStringQueryOptionDsl {
     fun text(value: String, field: String? = null): Query {
         val escaped = value.replace("*", "\\*").replace("?", "\\?")
 
-        return Query("\"$escaped\"", field)
+        query = Query("\"$escaped\"", field)
+
+        return query
```

Comment:
I think this idea is great. 
but there is another solution.

we can change `QueryStringSearchOperatorDsl.kt` like this.

```kt
    fun query(configuration: QueryStringQueryOptionDsl.() -> QueryStringQueryOptionDsl.Query) {
        val query = QueryStringQueryOptionDsl().configuration()
        document["query"] = query.toString()
    }
```

We no longer need to reassign the query in every function by using this method.
what do you think?

Reply:
Thanks for the suggestion.
I'm going to ask you again just to make sure I understand what you mean.

If I write it as you say, 
the `query`property that existed in the `QueryStringQueryOptionDsl`class will disappear?

Reply:
Yes, and build method also removed.